dependencies.go

```diff
@@ -24,7 +27,17 @@ const dependenciesKey key = iota
 
 type Dependencies interface {
 	Dependency
-	HTTPClient() (http.Client, error)
+	// HTTPClient returns both a HTTP client for requests, and a reader factory function
+	// that must be used when reading the response body. This may add some security.
+	// Example usage:
+	// ```
+	// deps := flux.GetDependencies(ctx)
+	// client, reader, _ := deps.HTTPClient()
+	// response, _ := client.Do(req)
+	// r := reader(response.Body)
+	// ...
+	// ```
+	HTTPClient() (http.Client, func(body io.Reader) io.Reader, error)
```

Comment:
It seems to me this can be done by wrapping the `http.Client` itself or modifying the `RoundTripper`.

```
func LimitHTTPBody(client http.Client, size int) http.Client {
    // The client is already a struct so it was already copied
    // which makes this safe.
    if client.Transport == nil {
        client.Transport = http.DefaultTransport
    }
    client.Transport = roundTripLimiter{RoundTripper: client.Transport, max: 1024}
    return client
}

type roundTripLimiter struct {
    http.RoundTripper
    max int
}

func (l roundTripLimiter) RoundTrip(r *http.Request) (*http.Response, error) {
    response, err := l.RoundTripper.RoundTrip(r)
    if err != nil {
        return nil, err
    }
    response.Body = newLimitReader(response.Body, l.max)
    return response, nil
}
```

This method also means that no other code needs to change and I think this does the same thing.

Reply:
Updating, thank you @jsternberg